src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,873 @@ def test_rotate_img_metadata(self):
+        """
+        data = model.DataArray(self.arr1, metadata=self.md)
+        # take a random z-slice value
+        z_slice = 160
+
+        output_DA = numpy.asarray(img.getYXFromZYX(data, zIndex=z_slice))
+
+        # check if dimensions of the returned ndarray
+        # are smaller than input array (ZYX vs YX)
+        self.assertTrue(output_DA.ndim < data.ndim)
+        # check if the XY slice output is part of input DA
+        self.assertTrue(output_DA in data)
+
+        # check if the slice has the same size (x,y) as input
+        self.assertEqual(output_DA.shape[0], data.shape[1])
+        self.assertEqual(output_DA.shape[1], data.shape[2])
+
+        # calculates a checknum value based on predicted values e.g. 3x3 = 9 elements containing value of color_num
+        # this value is based on input z_slice of 198 compared to max z value of 199 (array z shape - 1)
+        z_num = self.arr1.shape[0] - 1
+        checknum = ((1 + (z_num - z_slice)) + z_num - z_slice) * ((1 + (z_num - z_slice)) + z_num - z_slice)
+
+        self.assertTrue(numpy.count_nonzero(output_DA == self.color_num) == checknum)
+
+    def test_get_yx_from_zyx_1D(self):
+        """
+        Verify that using a 1D DA picture will return the
+        right 2D DA picture.
+        see if expected input of ZYX DA can handle a X only DA
+        """
+        # create a 2D data array using a sliced 3D array
+        # keep the z-slice at default
+        data = model.DataArray(self.arr1[0, 0, :], metadata=self.md)
+        output_DA = numpy.asarray(img.getYXFromZYX(data))
+
+        # check if the XY slice output is part of input DA
+        self.assertTrue(output_DA in data)
+        # check if the slice has the same size (x,y) as input
+        self.assertEqual(output_DA.shape[1], data.shape[0])
+        # check if all elements are of the set color_num
+        self.assertTrue(numpy.all(output_DA == self.color_num))
+
+    def test_get_yx_from_zyx_4D(self):
+        """
+        Verify that using a 4D DA picture will fail.
+        see if a ValueError exception is raised
+        """
+        # use a 4D array to force a ValueError when slicing array shape
+        data = model.DataArray(self.arr1.reshape(20, 399, 399, 10), metadata=self.md)
+
+        with self.assertRaises(ValueError):
+            img.getYXFromZYX(data)
+
+
+class TestEnsure2DImage(unittest.TestCase):
+    # input data array
+    # output 2D data array
+    # test for different dimension values
+    # test forcing ValueError
+
+    def setUp(self):
+        y = 4000
+        x = 4000
+        # size of picture in CTZYX
+        size = (1, 1, 1, y, x)
+        arr = numpy.ones(size, dtype='uint8')
+        self.data = model.DataArray(arr)
+
+    def test_ensure_2d_image_2d(self):
+        """
+        Verify that using a 5D CTZYX DA will return the
+        right 2D DA.
+        normal use of function
+        """
+        out_arr = numpy.asarray(img.ensure2DImage(self.data))
```

Comment:
Why using `numpy.asarray()`? Shouldn't you check directly the output of the function? (actually, I'm not sure whether `asarray()` has any effect here).